src/common/librlist/rlist.c

```diff
@@ -1271,17 +1436,117 @@ static json_t *rlist_json_nodelist (struct rlist *rl)
     return o;
 }
 
+static zhashx_t *rlist_properties (struct rlist *rl)
+{
+    int saved_errno;
+    struct rnode *n;
+    zlistx_t *keys = NULL;
+    zhashx_t *properties = zhashx_new ();
+
+    if (!properties)
+        return NULL;
+
+    zhashx_set_destructor (properties, property_destructor);
+
+    n = zlistx_first (rl->nodes);
+    while (n) {
+        if (n->properties) {
+            const char *name;
+            if (!(keys = zhashx_keys (n->properties)))
+                goto error;
+            name = zlistx_first (keys);
+            while (name) {
+                struct idset *ids = zhashx_lookup (properties, name);
+                if (!ids) {
+                    if (!(ids = idset_create (0, IDSET_FLAG_AUTOGROW)))
+                        goto error;
+                    if (zhashx_insert (properties, name, ids) < 0) {
+                        idset_destroy (ids);
+                        goto error;
+                    }
+                }
+                if (idset_set (ids, n->rank) < 0)
+                    goto error;
+                name = zlistx_next (keys);
+            }
+            zlistx_destroy (&keys);
+            keys = NULL;
+        }
+        n = zlistx_next (rl->nodes);
+    }
+
+    return properties;
+error:
+    saved_errno = errno;
+    zhashx_destroy (&properties);
+    if (keys)
+        zlistx_destroy (&keys);
+    errno = saved_errno;
+    return NULL;
+}
+
+static json_t *rlist_json_properties (struct rlist *rl)
+{
+    int saved_errno;
+    zhashx_t *properties = NULL;;
+    json_t *o = NULL;
+    json_t *result = NULL;
+    struct idset *ids;
+
+    if (!(properties = rlist_properties (rl)))
+        return NULL;
+
+    /*  Do not bother returning an empty JSON object
+     */
+    if (zhashx_size (properties) == 0)
+        goto out;
+
+    if (!(o = json_object ())) {
+        errno = ENOMEM;
+        goto out;
+    }
+    ids = zhashx_first (properties);
+    while (ids) {
+        const char *name = zhashx_cursor (properties);
+        char *s = idset_encode (ids, IDSET_FLAG_RANGE);
+        json_t *val;
+        if (!s || !(val = json_string (s))) {
```

Comment:
set `errno = ENOMEM` before `goto out`

Reply:
I _think_ this was fixed by 555ff3251e4918cfe8170c31190a134ed698dec3, but the line numbers referenced here don't match what is in the actual branch so maybe I'm confused...

Reply:
Yeah that's what was confusing me too.  I think we're good.

Reply:
This may be my fault. I introduced the bug in one commit and fixed in another. However, in this particular case that was cleaner because initially errors were ignored in the calling function. The follow-up commit fixed that along with ignored errors from the similar function that generates the `nodelist` key. Sorry about that! I'll try to refrain from doing that in the future.